c/json.c

```diff
@@ -322,11 +310,11 @@ writeBufferWithEscaping(jsonPrinter *p, size_t len, char text[len]) {
 
     UTF8_GET_NEXT_CHAR(len, text, &i, &utf8Char, &getCharErr);
     if (getCharErr != 0) {
-      JSONERROR("JSON: invalid UTF-8, rc %d\n", getCharErr);
+      zowelog(NULL, LOG_COMP_JSON, ZOWE_LOG_SEVERE, "JSON: invalid UTF-8, rc %d\n", getCharErr);
```

Comment:
I'm not sure if library code should start spitting out messages like this. I think, that in general, it's best to pass  info to the caller and the caller would decide if/what to print. Also, we don't know if after these changes, an error in JSON can cause log flooding.

One other thing is, we don't have message IDs for non-debug messages here. It's not a request to change this PR, but we should consider improving the messages here in the future.